account_payment_partner/models/account_move.py

```diff
@@ -65,23 +65,35 @@ def _compute_payment_mode_id(self):
                 move.payment_mode_id = False
             if move.partner_id:
                 partner = move.with_company(move.company_id.id).partner_id
-                if move.move_type == "in_invoice":
+                if move.move_type == "in_invoice" and partner.supplier_payment_mode_id:
                     move.payment_mode_id = partner.supplier_payment_mode_id
-                elif move.move_type == "out_invoice":
+                elif (
+                    move.move_type == "out_invoice" and partner.customer_payment_mode_id
+                ):
                     move.payment_mode_id = partner.customer_payment_mode_id
                 elif (
                     move.move_type in ["out_refund", "in_refund"]
                     and move.reversed_entry_id
+                    and move.reversed_entry_id.payment_mode_id
```

Comment:
No need of this line (and the same for the other 2 similar ones in following diffs)

Reply:
Lo reviso gracias.

Reply:
Done.